### PR TITLE
Document the 'checked build' feature

### PR DESCRIPTION
**Pull request by xmcclure**

### CODE REVIEW

**Comment by kumpera** [comment]
LGTM

File: docs/advanced/checked-build.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: Checked Builds
+redirect_from:
+  - /Checked_Build/
```
Comment (akoeplinger) [inline docs/advanced/checked-build.md]: You don't need those redirects for new content, the top level URLs are a relic of the old MediaWiki based website.

File: docs/advanced/checked-build.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+Introduction
+============
+
+The Mono VM uses asserts throughout to ensure it is in a correct state. There are a few classes of asserts which are useful for debugging Mono, but which are too expensive to include in a production VM. If you are doing development on Mono, you can enable these with a specially built VM.
+
+To enable the checked modes, you must build with the `--with-checked-build=` argument passed to `autogen.sh`/`configure`, and then run with the `MONO_CHECK_MODE` environment variable. Both `--with-checked-build` and `MONO_CHECK_MODE` should be set to either the word `all`, or a comma-separated list of mode names if you need to do fine-grained testing. (In particular you might want to skip `metadata` mode, since it is exceptionally slow.)
```
Comment (akoeplinger) [inline docs/advanced/checked-build.md]: The configure flag is `--enable-checked-build=` after your PR. This applies throughout the doc.

Comment (akoeplinger) [inline docs/advanced/checked-build.md]: Why do I need to set both the configure flag and the env var to the same list? What happens if they don't match?

File: docs/advanced/checked-build.md (hunk 1 above)
Comment (xmcclure) [inline docs/advanced/checked-build.md]: What URL do you get if you don't include a redirect?

Comment (xmcclure) [inline docs/advanced/checked-build.md]: Oh, wait, I think I get it.

Comment (akoeplinger) [inline docs/advanced/checked-build.md]: The page is always at http://www.mono-project.com/docs/advanced/checked-build/, if you add the redirect_from it'll just additionally redirect there from http://www.mono-project.com/Checked_Build/

**Comment by xmcclure** [comment]
https://github.com/mono/mono/pull/2753 is going in as soon as a test passes so merging this.

**Pull request merged**